src/net.c

```diff
@@ -3144,6 +3161,28 @@ coap_can_exit(coap_context_t *context) {
   }
   return 1;
 }
+#ifndef WITHOUT_ASYNC
+coap_tick_t
+coap_check_async(coap_context_t *context, coap_tick_t now) {
+  coap_tick_t next_due = 0;
+  coap_async_t *async, *tmp;
+
+  LL_FOREACH_SAFE(context->async_state, async, tmp) {
+    if (async->delay <= now) {
+      /* Send off the request to the application */
+      handle_request(context, async->session, async->pdu);
```

Comment:
I wonder if this behavior is useful. Separate responses are used by applications that may not know in advance how long it takes to create the response (because of long-taking sensor readings, or network messages send back and forth, etc.). With the API proposed here, this type of application would need to guess a reasonable delay and periodically call `coap_async_set_delay()` to extend it until it is able to send the actual response to the CoAP request. Once it fails to extend the delay, the async object gets removed.

I see two ways to overcome this limitation: (1) would be to extend `coap_response_t` to indicate that no final response has been determined yet. This has the drawback that the async information always needs to be retrieved. (2) A callback function is registered in addition to the delay value. Once the delay has expired, the callback function is called with the async structure as its argument. The callback function returns a new delay value when no response is ready yet, or $somespecialvalue that indicates if the request handler should be called to retrieve the response or abort the operation.

Reply:
It is for this reason that you can set `delay `to 0 in `coap_register_async()` - meaning never trigger, and then when everything is ready, call `coap_async_set_delay()` with a small value - e.g. 1 which then causes the request handler to be immediately called (similar to how `coap_resource_notify_observers()` is used to trigger a response).

I agree that I did not pull this out in the documentation.

It is also my intent to use this type of logic to delay a multicast response as per RFC7252.

Reply:
Added in a note to describe this usage - code pushed.

Reply:
Okay, this makes sense.